src/library/filescanner_ffmpeg.c

```diff
@@ -295,44 +310,64 @@ extract_metadata_core(struct media_file_info *mfi, AVDictionary *md, const struc
     DPRINTF(E_DBG, L_SCAN, " -> %s = %s\n", mdt->key, mdt->value);
 #endif
 
-  mdcount = 0;
-
-  /* Extract actual metadata */
-  for (i = 0; md_map[i].key != NULL; i++)
+  mdcount = 0;    
+  
+  /* Cache this search once to avoid doing it per metadata key */
+  lyricsindex = match_metadata("lyrics", md_map);
+  
+  /* Extract lyrics if any found. 
+     FFMPEG creates a metadata key that's embedding the language code in it, like 'lyrics-eng' or 'lyrics-XXX'
+     So it's not possible to query this key directly except by brute-forcing all languages.
+     Instead, we are reversing the metadata searching algorithm to query all metadata key that FFMPEG fetched 
+     and matching them against our own map. This is the most efficient method to search it without having 2 pass
+     on the KV store */
+  mdt = av_dict_get(md, "", NULL, AV_DICT_IGNORE_SUFFIX); /* from av_dict_get documentation */
+  while (mdt != NULL)
     {
-      mdt = av_dict_get(md, md_map[i].key, NULL, 0);
-      if (mdt == NULL)
-	continue;
-
-      if ((mdt->value == NULL) || (strlen(mdt->value) == 0))
-	continue;
-
-      if (md_map[i].handler_function)
-	{
-	  mdcount += md_map[i].handler_function(mfi, mdt->value);
-	  continue;
-	}
-
-      mdcount++;
-
-      if (!md_map[i].as_int)
-	{
-	  strval = (char **) ((char *) mfi + md_map[i].offset);
-
-	  if (*strval == NULL)
-	    *strval = strdup(mdt->value);
-	}
-      else
-	{
-	  intval = (uint32_t *) ((char *) mfi + md_map[i].offset);
-
-	  if (*intval == 0)
-	    {
-	      ret = safe_atou32(mdt->value, intval);
-	      if (ret < 0)
-		continue;
```

Comment:
I'm pretty sure it's a bug here. If it failed converting to int, shouldn't mdcount be decremented ?

Reply:
Yes, just looks like the mdcount++ should have been placed lower (it just seems to be used for logging, however).

Reply:
Ok, will change this tomorrow so it doesn't return 1 but 0 in that case.